### PR TITLE
add pvc-autoresizer controller to ks-controller-manager

### PR DESCRIPTION
**Pull request by f10atin9**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:
Resizing pvc automatically is a good idea, and the additional installation of pvc-autoresizer is not convenient enough.
So I add pvc-autoresizer's controller to ks. There is no need to install pvc-autoresizer additionally.

### Which issue(s) this PR fixes:

Fixes #
#4185 
### Special notes for reviewers:
```
None
```

### Does this PR introduced a user-facing change?

```release-note
When the user needs to use these functions, they just need to add the same annotations as before with pvc-autoresizer.
The console will support control it soon.
```

### Additional documentation, usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: cmd/controller-manager/app/controllers.go [hunk 1]
```diff
@@ -348,6 +352,36 @@ func addAllControllers(mgr manager.Manager, client k8s.Client, informerFactory i
 		addController(mgr, "volumesnapshot", volumeSnapshotController)
 	}
 
+	// "pvc-autoresizer"
+	if cmOptions.IsControllerEnabled("pvc-autoresizer") {
+		if err := runners.SetupIndexer(mgr, false); err != nil {
+			return err
+		}
+
+		promClient, err := runners.NewPrometheusClient("http://prometheus-k8s.kubesphere-monitoring-system:9090")
```
Comment (stoneshi-yunify) [inline cmd/controller-manager/app/controllers.go:361]: is the url `http://prometheus-k8s.kubesphere-monitoring-system:9090` fixed? hard code is not a good idea. at least it should be constructed with some constants.

**Review by stoneshi-yunify — CHANGES_REQUESTED** [review]
overall looks good. good job!

File: cmd/controller-manager/app/controllers.go (hunk 1 above)
Comment (f10atin9) [reply cmd/controller-manager/app/controllers.go:361]: Okay，I‘ll fix it.

**Review by stoneshi-yunify — APPROVED** [review]
/lgtm

File: cmd/controller-manager/app/controllers.go [hunk 2]
```diff
@@ -348,6 +354,37 @@ func addAllControllers(mgr manager.Manager, client k8s.Client, informerFactory i
 		addController(mgr, "volumesnapshot", volumeSnapshotController)
 	}
 
+	// "pvc-autoresizer"
+	if cmOptions.IsControllerEnabled("pvc-autoresizer") {
+		if err := runners.SetupIndexer(mgr, false); err != nil {
+			return err
+		}
+
+		prometheusUrl := "http://" + constants.PrometheusSvcName + "." + constants.KubeSphereMonitoringNamespace + ":" + constants.PrometheusSvcPort
```
Comment (zryfish) [inline cmd/controller-manager/app/controllers.go:363]: Shouldn't create prometheusUrl using default service name, the endpoint of Prometheus could be different. Please use the address of  monitoring endpoint, you can reference https://github.com/kubesphere/kubesphere/blob/master/cmd/ks-apiserver/app/options/options.go#L123 for example.

File: cmd/controller-manager/app/controllers.go [hunk 3]
```diff
@@ -348,6 +352,38 @@ func addAllControllers(mgr manager.Manager, client k8s.Client, informerFactory i
...
+				mgr.GetClient(),
+				ctrl.Log.WithName("pvc-autoresizer"),
+				1*time.Minute,
+				mgr.GetEventRecorderFor("pvc-autoresizer"),
+			)
+			addController(mgr, "pvcautoresizer", pvcAutoResizerController)
+		}
+	}
+
+	if cmOptions.IsControllerEnabled("workload-Restart") {
```
Comment (zryfish) [inline cmd/controller-manager/app/controllers.go:377]: `workload-Restart` is a bad controller name, and its style is inconsistent, maybe choose another name.

Comment (f10atin9) [reply cmd/controller-manager/app/controllers.go:377]: How about `expansion-restarter` or`autorestarter`?  Or any other better suggestions?

Comment (zryfish) [reply cmd/controller-manager/app/controllers.go:377]: yeah, sounds better. This controller is only used when PVC size changes, right? So, how about `pvc-workload-restarter`?

Comment (f10atin9) [reply cmd/controller-manager/app/controllers.go:377]: Yes it does, `pvc-workload-restarter` sounds great, I'll update it, and thanks for the suggestion.

**Pull request merged**